core/domain/state_domain_test.py

```diff
@@ -4443,6 +4444,321 @@ def test_traverse_by_schema_and_get(self):
         self.assertEqual(html, ['<p>testing</p>'])
 
 
+class InteractionInstanceDomainTests(test_utils.GenericTestBase):
+
+    def test_to_proto(self):
```

Comment:
Is this test only for checking the type? If so, say that in the name.

Though shouldn't you be verifying the properties of the proto formed by to_proto()?

You should probably also have a test for "if any interaction not listed here has to_proto() called, a failure arises" and "all interactions defined in oppia fit in either that test or this one". That way, when someone adds a new interaction, they will be forced to make sure it is accounted for here.

I noticed above that there's a "verify_interaction_supports_android" so perhaps another alternative is to ensure (programmatically) that all android-supported interactions have working to_proto() methods and all non-android-supported interactions do not; the latter part should be checked based on running through a list rather than hardcoding a test for each non-android-supported interaction (again so that if someone adds a new interaction, they are forced to ensure it's accounted for somehow in these tests).

Reply:
Added more check in  the same test.

Reply:
Added a check if an interaction has to_proto function or not. 

```py
        for interaction_id in android_validation_constants.VALID_INTERACTION_IDS:
            interaction_obj = state_domain.InteractionInstance(
                interaction_id,
                {}, [], default_outcome, [], [], None)
            if interaction_obj.is_supported_on_android_app():
                self.assertTrue(
                    hasattr(interaction_obj, 'to_proto')
                )
            else:
                self.assertFalse(
                    hasattr(interaction_obj, 'to_proto')
                )
```